models/choral.js

```diff
@@ -125,6 +125,89 @@ choralSchema.statics.findDevicesForUser = function (user, cb) {
   });
 };
 
+choralSchema.statics.findTreeForUser = function (user, cb) {
+  this.find({ userId: user._id }, (err, chorals) => {
+    if (err) return cb(err, null);
+
+    var all = {};
+    var children = [];
+    for (i in chorals) {
```

Comment:
use for(var i in chorals) instead of declaring global.

Also since you're iterating through object, maybe change "i" to "choral" instead, it makes it more clear that it's an object key when it's used to index the object